src/Processors/Formats/Impl/ORCBlockOutputFormat.cpp

```diff
@@ -0,0 +1,441 @@
+        }
+        case TypeIndex::UInt64: [[fallthrough]];
+        case TypeIndex::Int64:
+        {
+            return orc::createPrimitiveType(orc::TypeKind::LONG);
+        }
+        case TypeIndex::Float32:
+        {
+            return orc::createPrimitiveType(orc::TypeKind::FLOAT);
+        }
+        case TypeIndex::Float64:
+        {
+            return orc::createPrimitiveType(orc::TypeKind::DOUBLE);
+        }
+        case TypeIndex::Date:
+        {
+            return orc::createPrimitiveType(orc::TypeKind::DATE);
+        }
+        case TypeIndex::DateTime: [[fallthrough]];
+        case TypeIndex::DateTime64:
+        {
+            return orc::createPrimitiveType(orc::TypeKind::TIMESTAMP);
+        }
+        case TypeIndex::FixedString: [[fallthrough]];
+        case TypeIndex::String:
+        {
+            return orc::createPrimitiveType(orc::TypeKind::BINARY);
+        }
+        case TypeIndex::Nullable:
+        {
+            return getORCType(removeNullable(type));
+        }
+        case TypeIndex::Array:
+        {
+            const auto * array_type = typeid_cast<const DataTypeArray *>(type.get());
+            return orc::createListType(getORCType(array_type->getNestedType()));
+        }
+        case TypeIndex::Decimal32:
+        {
+            const auto * decimal_type = typeid_cast<const DataTypeDecimal<Decimal32> *>(type.get());
+            return orc::createDecimalType(decimal_type->getPrecision(), decimal_type->getScale());
+        }
+        case TypeIndex::Decimal64:
+        {
+            const auto * decimal_type = typeid_cast<const DataTypeDecimal<Decimal64> *>(type.get());
+            return orc::createDecimalType(decimal_type->getPrecision(), decimal_type->getScale());
+        }
+        case TypeIndex::Decimal128:
+        {
+            const auto * decimal_type = typeid_cast<const DataTypeDecimal<Decimal128> *>(type.get());
+            return orc::createDecimalType(decimal_type->getPrecision(), decimal_type->getScale());
+        }
+        default:
+        {
+            throw Exception("Type " + type->getName() + " is not supported for ORC output format", ErrorCodes::ILLEGAL_COLUMN);
+        }
+    }
+}
+
+template <typename NumberType, typename NumberVectorBatch>
+void ORCBlockOutputFormat::ORCBlockOutputFormat::writeNumbers(
+        orc::ColumnVectorBatch * orc_column,
+        const IColumn & column,
+        const PaddedPODArray<UInt8> * null_bytemap)
+{
+    NumberVectorBatch * number_orc_column = dynamic_cast<NumberVectorBatch *>(orc_column);
+    const auto & number_column = assert_cast<const ColumnVector<NumberType> &>(column);
+    number_orc_column->resize(number_column.size());
+
+    for (size_t i = 0; i != number_column.size(); ++i)
+    {
+        if (null_bytemap && (*null_bytemap)[i])
+        {
+            number_orc_column->notNull[i] = 0;
+            continue;
+        }
+        if constexpr (std::is_same<NumberType, UInt8>::value)
```

Comment:
I think it is dirty hack :) Please write comment about this special case.

Reply:
And it doesn't work https://clickhouse-builds.s3.yandex.net/11662/717f63923a19b9a4b0932e6295ed60d3753e7009/build_log_707272824_1592914485.txt
Maybe add `Int8` to if statement?